pkg/contracts/sui/ptb_argument.go

```diff
@@ -0,0 +1,58 @@
+package sui
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pattonkan/sui-go/sui"
+	"github.com/pattonkan/sui-go/sui/suiptb"
+	"github.com/pkg/errors"
+)
+
+// PureUint64ArgFromStr converts a string to a uint64 and creates a PTB pure argument
+func PureUint64FromString(
+	ptb *suiptb.ProgrammableTransactionBuilder,
+	integerStr string,
+) (arg suiptb.Argument, value uint64, err error) {
+	value, err = strconv.ParseUint(integerStr, 10, 64)
+	if err != nil {
+		return suiptb.Argument{}, 0, errors.Wrapf(err, "failed to parse amount %s", integerStr)
+	}
+
+	arg, err = ptb.Pure(value)
+	if err != nil {
+		return suiptb.Argument{}, 0, errors.Wrapf(err, "failed to create amount argument")
+	}
+
+	return arg, value, nil
+}
+
+// TypeTagFromString creates a PTB type argument StructTag from a type string
+// Example: "0000000000000000000000000000000000000000000000000000000000000002::sui::SUI" ->
+//
+//	&sui.StructTag{
+//		Address: "0x0000000000000000000000000000000000000000000000000000000000000002",
+//		Module:  "sui",
+//		Name:    "SUI",
+//	}
+func TypeTagFromString(t string) (*sui.StructTag, error) {
```

Comment:
Let's remove the unnecessary pointer.
```suggestion
func TypeTagFromString(t string) (sui.StructTag, error) {
```

Reply:
https://github.com/zeta-chain/node/pull/3793/commits/551618572f01a1d718ea62a944a79b049ff02360